scripts/volume-recovery

```diff
@@ -0,0 +1,147 @@
+            usage
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+mandatory=(
+  SNAPSHOT
+  INSTANCE_ID
+  OLD_VOLUME
+)
+for variable in ${mandatory[@]}; do
+  if [[ -z ${variable} ]]; then
+    echo "Mandatory variable missing: [${BOLD}${variable}${NORMAL}]"
+    exit 1
+  fi
+done
+
+# Check old volume and instance exist together
+VOLUME_CHECK_JSON=$(aws ec2 describe-volumes --volume-ids ${OLD_VOLUME})
+# Checking snapshot start time for validation
+SNAPSHOT_CHECK_JSON=$(aws ec2 describe-snapshots --snapshot-ids ${SNAPSHOT})
+# Bring back the instance name which is in the Name key tag
+INSTANCE_CHECK_JSON=$(aws ec2 describe-instances --instance-id ${INSTANCE_ID})
+
+INSTANCE_NAME=$(echo ${INSTANCE_CHECK_JSON} | jq -r '.Reservations[].Instances[].Tags[] | select(.Key == "Name").Value')
+INSTANCE_REGION_JSON=$(echo ${VOLUME_CHECK_JSON} | jq -r .Volumes[].AvailabilityZone)
+SNAPSHOT_START_TIME=$(echo ${SNAPSHOT_CHECK_JSON} | jq -r .Snapshots[].StartTime)
+VOLUME_INSTANCE_ID=$(echo ${VOLUME_CHECK_JSON} | jq -r .Volumes[].Attachments[].InstanceId)
+if [[ ${VOLUME_INSTANCE_ID} != ${INSTANCE_ID} ]]; then
+  INSTANCE_NAME=$(aws ec2 describe-instances --instance-id ${VOLUME_INSTANCE_ID} | jq -r '.Reservations[].Instances[].Tags[] | select(.Key == "Name").Value')
+  echo "Volume is not attached to this instance, it appears to be attached to ${INSTANCE_NAME} (${VOLUME_INSTANCE_ID})"
+  exit 1
+fi
+
+echo "Snapshot is ${SNAPSHOT}, which was created ${SNAPSHOT_START_TIME%.*}"
+echo "Instance is ${INSTANCE_ID}, the name is ${INSTANCE_NAME}"
+echo "Old volume is ${OLD_VOLUME}, which is in ${INSTANCE_REGION_JSON}"
+
+read -p "Do you want to proceed? y or n " -n 1 -r
+echo
+if [[ ! $REPLY =~ ^[Yy]$ ]]
+then
+    exit 1
+fi
+
+OLD_VOLUME_JSON=$(aws ec2 describe-volume-status --volume-ids ${OLD_VOLUME})
+INSTANCE_JSON=$(aws ec2 describe-instance-status --instance-id ${INSTANCE_ID} --include-all-instances)
+
+# Create volume from snapshot and wait for an 'ok' status
+echo "Creating new volume from ${SNAPSHOT}"
+VOLUME=$(aws ec2 create-volume --volume-type gp2 --snapshot-id ${SNAPSHOT} --availability-zone ${INSTANCE_REGION_JSON})
+VOLUME_ID=$(echo ${VOLUME} | jq -r .VolumeId)
+echo "Volume created is ${VOLUME_ID}"
+aws ec2 create-tags --resources ${VOLUME_ID} --tags Key=Snapshot,Value=daily Key=Name,Value=devops-prometheus-instance1
+VOLUME_JSON=$(aws ec2 describe-volume-status --volume-ids ${VOLUME_ID})
+VOLUME_STATUS=$(echo ${VOLUME_JSON} | jq -r .VolumeStatuses[].VolumeStatus.Status)
+until [[ ${VOLUME_STATUS} == "ok" ]]; do
+  sleep 5
+  VOLUME_JSON=$(aws ec2 describe-volume-status --volume-ids ${VOLUME_ID})
+  VOLUME_STATUS=$(echo ${VOLUME_JSON} | jq -r .VolumeStatuses[].VolumeStatus.Status)
+  echo ${VOLUME_STATUS}
+done
+echo "Success"
+
+# Stop the EC2 instance and wait for a 'stopped' status
+echo "Shutting down ${INSTANCE_NAME}"
+INSTANCE=$(aws ec2 stop-instances --instance-ids ${INSTANCE_ID})
```

Comment:
I'd change this to `INSTANCE_STOP=...` similar to the **_start_** one you have below. Doesn't look like the variable is used anywhere at  the mo, so I guess there's an argument for dropping the variable assignment completely - but that's probably more of a change than you want to do today!

Reply:
Suspect what we should probably do as a future enhancement is check the `INSTANCE_STOP` output for success too.